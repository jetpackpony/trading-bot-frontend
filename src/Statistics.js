import React from 'react';
import moment from 'moment';
import * as R from 'ramda';

const fractionToPercent =
  (num) => Math.abs(Math.round(num * 10000) / 100);
const getSign =
  (num) => (
    (num > 0)
    ? '+'
    : (num < 0)
    ? '-'
    : ''
  );
const joinVariadic = R.unapply(R.join(''));
const composePercentage =
  R.converge(
    joinVariadic,
    [getSign, fractionToPercent, R.always('%')]
  );

const StatLine =
  ({
    label,
    value
  }) => (
    <div className="row">
      <div className="col-xs-6">{label}</div>
      <div className="col-xs-6">{value}</div>
    </div>
  );

const Statistics =
  ({
    startAt,
    profitTotal,
    profit24,
    currentDealProfit
  }) => (
    <div className="Statistics row">
      <StatLine
        label="Started"
        value={ moment(startAt).format('LLLL') }
      />
      <StatLine
        label="Profit since start:"
        value={ composePercentage(profitTotal) }
      />
      <StatLine
        label="Profit in the last 24h:"
        value={ composePercentage(profit24) }
      />
      { (currentDealProfit !== null) ? (
        <StatLine
          label="Currently open deal:"
          value={ composePercentage(currentDealProfit) }
        />
      ) : (
        <div className="row">
          <div className="col-xs-12">No open deals, waiting for the opportunity</div>
        </div>
      ) }
    </div>
  );

export default Statistics;
