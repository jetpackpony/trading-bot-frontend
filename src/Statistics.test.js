import React from 'react';
import Statistics from '../src/Statistics';
import renderer from 'react-test-renderer';

test('Statistics shows current deal if it is set', () => {
  const comp = renderer.create(
    <Statistics
      startAt={1523030489163}
      profitTotal={0.143245}
      profit24={-0.01124}
      currentDealProfit={0.15}
    />
  );

  let tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Statistics shows that no deal is open', () => {
  const comp = renderer.create(
    <Statistics
      startAt={1523030489163}
      profitTotal={0.243245}
      profit24={-0.01124}
      currentDealProfit={null}
    />
  );

  let tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});
