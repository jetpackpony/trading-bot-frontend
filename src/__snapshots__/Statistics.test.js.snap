// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics shows current deal if it is set 1`] = `
<div
  className="Statistics row"
>
  <div
    className="row"
  >
    <div
      className="col-xs-6"
    >
      Started
    </div>
    <div
      className="col-xs-6"
    >
      Friday, April 6, 2018 7:01 PM
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-6"
    >
      Profit since start:
    </div>
    <div
      className="col-xs-6"
    >
      +14.32%
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-6"
    >
      Profit in the last 24h:
    </div>
    <div
      className="col-xs-6"
    >
      -1.12%
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-6"
    >
      Currently open deal:
    </div>
    <div
      className="col-xs-6"
    >
      +15%
    </div>
  </div>
</div>
`;

exports[`Statistics shows that no deal is open 1`] = `
<div
  className="Statistics row"
>
  <div
    className="row"
  >
    <div
      className="col-xs-6"
    >
      Started
    </div>
    <div
      className="col-xs-6"
    >
      Friday, April 6, 2018 7:01 PM
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-6"
    >
      Profit since start:
    </div>
    <div
      className="col-xs-6"
    >
      +24.32%
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-6"
    >
      Profit in the last 24h:
    </div>
    <div
      className="col-xs-6"
    >
      -1.12%
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      No open deals, waiting for the opportunity
    </div>
  </div>
</div>
`;
